### PR TITLE
tests: reduce checkpoint experiment iterations

### PR DESCRIPTION
**Pull request by pmrowla**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Cuts `tests/func/experiments` from 137s to 84s on my machine

### CODE REVIEW

**Comment by pmrowla** [comment]
Still need further investigation into whether we can optimize the multiprocessing overhead for experiment executors, but in the meantime this should still help

File: dvc/api/__init__.py [hunk 1]
```diff
@@ -139,4 +139,4 @@ def make_checkpoint():
         fobj.flush()
         os.fsync(fobj.fileno())
     while os.path.exists(signal_file):
-        sleep(1)
+        sleep(0.1)
```
Comment (efiop) [inline dvc/api/__init__.py:142]: we could move this into a constant and patch it only for tests. Kinda like we do with moto https://github.com/iterative/dvc/blob/08300d3bcf51f7bd9c8509924110416461b11b60/tests/func/test_s3.py#L13

**Review by pared — APPROVED** [review]

File: dvc/api/__init__.py (hunk 1 above)
Comment (pmrowla) [reply dvc/api/__init__.py:142]: Dmitry has mentioned before that he thought this 1s sleep could be sped up in general, so this change doesn't really need to be for tests only.

Comment (efiop) [reply dvc/api/__init__.py:142]: Still better to not hardcode such things :slightly_smiling_face:

**Pull request merged**